crates/sui-indexer/src/store/module_resolver_v2.rs

```diff
@@ -1,38 +1,43 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::handlers::tx_processor::IndexingPackageCache;
-use crate::metrics::IndexerMetrics;
-use crate::schema_v2::packages;
-use crate::types_v2::IndexedPackage;
+use async_trait::async_trait;
 use diesel::ExpressionMethods;
-use diesel::QueryDsl;
-use diesel::RunQueryDsl;
+use diesel::OptionalExtension;
+use diesel::{QueryDsl, RunQueryDsl};
+use std::sync::{Arc, Mutex};
+
 use move_binary_format::CompiledModule;
 use move_bytecode_utils::module_cache::GetModule;
+use move_core_types::account_address::AccountAddress;
 use move_core_types::language_storage::ModuleId;
 use move_core_types::resolver::ModuleResolver;
-use std::sync::{Arc, Mutex};
-use sui_types::base_types::ObjectID;
+use sui_package_resolver::{error::Error as PackageResolverError, Package, PackageStore};
+use sui_types::base_types::{ObjectID, SequenceNumber};
 use sui_types::move_package::MovePackage;
+use sui_types::object::Object;
 
 use crate::errors::{Context, IndexerError};
+use crate::handlers::tx_processor::{IndexingModuleBuffer, IndexingPackageBuffer};
+use crate::metrics::IndexerMetrics;
 use crate::models_v2::packages::StoredPackage;
+use crate::schema_v2::{objects, packages};
 use crate::store::diesel_macro::read_only_blocking;
+use crate::types_v2::IndexedPackage;
 use crate::PgConnectionPool;
 
 /// A package resolver that reads packages from the database.
-pub struct IndexerStoreModuleResolver {
+pub struct IndexerStorePackageModuleResolver {
```

Comment:
nit: (Assuming was can get rid of the `ModuleResolver` impl) can we rename this as `IndexerStorePackageResolver`?

Reply:
great lead, following the usage of `module_resolver` I actually found more work needed on the reading path like
- make `try_into_object_read` and `try_into_dynamic_field_info` pkg upgrade aware
- impl. a package_cache to replace the current module_cache
I will put up a quick follow-up PR for this purpose.

Reply:
@amnn I found that another indexer usage of module_resolver other than `LayoutResolver` is for `get_signature_types`, naively this cannot be easily replaced by package_resolver or PackageStore, wdyt?
https://github.com/MystenLabs/sui/blob/main/crates/sui-json-rpc-types/src/sui_transaction.rs#L1622

Reply:
the indexer callsite is 
https://github.com/MystenLabs/sui/blob/main/crates/sui-indexer/src/models_v2/transactions.rs#L118

Reply:
It should be possible to replace this with similar logic that fetches a `FunctionDef` from a package resolver, but it will take a little bit more work (mostly to get the types to understand each other), but if I understand correctly, the main purpose of this code is to infer the types of pure inputs in transaction blocks, which means that although the current implementation will fail to find upgraded types in function signatures, it won't affect correctness, because we only care about the primitive types.

I do think we should still make that change, but mostly so we can eliminate `GetModule` for good -- so it is less urgent than getting the correctness fix out the door.

Reply:
follow up PR [#16042](https://github.com/MystenLabs/sui/pull/16042)
resolving this commend